akka-javasdk/src/main/scala/akka/javasdk/impl/eventsourcedentity/EventSourcedEntityEffectImpl.scala

```diff
@@ -91,8 +92,12 @@ private[javasdk] class EventSourcedEntityEffectImpl[S, E]
     this.asInstanceOf[EventSourcedEntityEffectImpl[T, E]]
   }
 
-  override def error[T](description: String): EventSourcedEntityEffectImpl[T, E] = {
-    _secondaryEffect = ErrorReplyImpl(description)
+  override def error[T](message: String): EventSourcedEntityEffectImpl[T, E] = {
+    error(new UserException(message))
```

Comment:
Clever!
When using `error(String)` we can still match it as exception if we want. Good idea!

Reply:
Or we can keep the `error(String)` behavior as it is, and stay with `error`?

Reply:
As discussed during the daily, currently `effect.error(String)` will be transformed into IllegalArgumentException, to be more backward compatible, UserException will also extend IllegalArgumentException.